File: problems_001to025/problem_001/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import natural_sum


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 3*natural_sum(999//3) + 5*natural_sum(999//5) - 15*natural_sum(999//15)
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_001/solution.ts
import { SolutionFunction } from '../../lib/types';
import { naturalSum } from '../../lib/toolbox';

const solution: SolutionFunction = () => {
  let answer = 0;

  answer += 3 * naturalSum(Math.floor(999 / 3));
  answer += 5 * naturalSum(Math.floor(999 / 5));
  answer -= 15 * naturalSum(Math.floor(999 / 15));

  return answer;
};

export default solution;
